lib/symboldatabase.cpp

```diff
@@ -4253,20 +4288,23 @@ bool Scope::isVariableDeclaration(const Token* const tok, const Token*& vartok,
     if (Token::Match(localVarTok, "%name% ;|=") || (localVarTok && localVarTok->varId() && localVarTok->strAt(1) == ":")) {
         vartok = localVarTok;
         typetok = localTypeTok;
+        return nullptr != vartok && vartok->isName();
```

Comment:
There were some circumstances where it would conclude something like "(" was a variable.

Reply:
I believe the test case will be exercised by removing the isName check again. I'll do so and document my findings.

As a general comment I often struggle to figure out if something is or is not a bug I think there's a lot of places where the design is forced to make an arbitrary decision (in the sense that either decision would be valid) and it's hard to know if the behavior that seems odd to me is by design or not. It would be useful for there to be a resource which documented the contract for some of these members more thoroughly (though I can see it would be tedious to write). If this resource exists already and I've just missed it please correct me.

Reply:
> If this resource exists already and I've just missed it please correct me.

there is no such resource as far as I know.

Reply:
Let me know if there is something in particular that you wonder about. And I can try to comment that better. A comment in the code would seems like a good place.

About varid.. as far as I remember that should only be set on variable names. Operators should never have a varId as far as I know.

We sometimes say `varId` and sometimes we say `declarationId`. If two tokens point at the same memory then varId should be the same. If the tokens have the same declaration then `declarationId` should be the same. Example:

    struct { int x; } s[10];
    void foo() {
        s[0].x = 1;
        s[1].x = 2;
    }

.. `s[0].x` and `s[1].x` have the same declaration. but they do not point at the same memory so `declarationId` should be the same in both expressions but `varId` should not be the same.

Now.. I guess it would be nice to put this info in a header somewhere.. not sure where.

Reply:
I'll have a go at documenting my understanding then and you can say if it's right or not.

Reply:
I wasn't able to reproduce the case I encountered where the name was not a valid name. As such I can't say for sure that it wasn't a result of something I was doing at the time or something I inadvertently fixed in some other way. Since it doesn't cause any current test failures I've reverted the name check as part of my refactoring in 1f09ab34da759cfe8e3ca2c948bb82639ae57e71